src/function/gds/gds_task.cpp

```diff
@@ -10,19 +10,19 @@ void FrontierTask::run() {
     auto numApproxActiveNodesForNextIter = 0u;
     std::unique_ptr<graph::Graph> graph = sharedState->graph->copy();
     auto state = graph->prepareScan(sharedState->relTableIDToScan);
-    auto scanResult = graph::GraphScanResult();
     auto localEc = sharedState->ec.copy();
     while (sharedState->frontierPair.getNextRangeMorsel(frontierMorsel)) {
         while (frontierMorsel.hasNextOffset()) {
             common::nodeID_t nodeID = frontierMorsel.getNextNodeID();
             if (sharedState->frontierPair.curFrontier->isActive(nodeID)) {
-                graph->scanFwd(nodeID, *state, scanResult);
-                for (auto i = 0u; i < scanResult.size(); ++i) {
-                    auto nbrNodeID = scanResult.nbrNodeIDs[i];
-                    auto edgeID = scanResult.edgeIDs[i];
-                    if (localEc->edgeCompute(nodeID, nbrNodeID, edgeID)) {
-                        sharedState->frontierPair.nextFrontier->setActive(nbrNodeID);
-                        numApproxActiveNodesForNextIter++;
+                for (const auto [nodes, edges] : graph->scanFwd(nodeID, *state)) {
```

Comment:
I just a minor question. In my future PRs, depends on configuration, I may need to both scanFwd and scanBwd.
Will the following works if I'm reusing the same state? Or do I need fwdState and bwdState?

```
for (const auto [nodes, edges] : graph->scanFwd(nodeID, *state)) {
}

for (const auto [nodes, edges] : graph->scanBwd(nodeID, *state)) {
}
```

Reply:
Yes, that should be fine. You also should always be able to re-use the state for a new scan if you've finished the original, you just can't nest them (even with a backward scan, though that would actually be fairly easy to allow as long as it's not recursive).